### ISSUE TITLE
Add option to adjust font size separately from GUI size.

### ISSUE THREAD

**Issue by YodasWs** [OpenTTD/OpenTTD#6918]
With respect to #6482, I want to bring this back up for discussion
> **Reason for closing: _Not a bug_**
> > please use the GUI zoom setting to increase sizes

On modern high-density screens, we need to be able to set the font size separately from the overall GUI size. If I increase the GUI zoom, then all the buttons and boxes are too large and block my view of playing the game. However, with the GUI zoom at 1×, the text on the minimap is a mere 4 pixels! That's too small to be able to read comfortably, especially when it's that light dim grey over that orangy-brown box.

**Comment by James103** [comment]
I agree. If you are playing OpenTTD at any resolution greater than 720p and 96 DPI, then the tiny font, which is 4 pixels at zoom 1x, is too small to read comfortably. But, if you set the GUI zoom at 4x, then you would need a 4K display in order to keep a similar window layout and to not block the main viewport too much. Although the tiny font would be at 16 pixels, the increased DPI that come from the high-res displays (even with the increased physical size) would negate the increased font size, and the tiny font would still be too small to read comfortably.

The options for the text size would be the same as the GUI zoom: 1x, 2x, and 4x.

**Comment by andythenorth** [comment]
I agree with the points, I'm using a high-DPI screen and my eyesight isn't what it once was.

But I'm unclear how text zoom can be achieved without UI zoom.  Lots of the UI chrome and widgets would need to be physically bigger to accommodate a bigger font :)

Meanwhile, you can configure your own fonts if that helps?
https://wiki.openttd.org/FAQ_troubleshooting#My_User_Interface_is_too_small_to_read_.2F_My_font_is_unreadable_or_faulty

cheers,

Andy

**Comment by James103** [comment]
So basically, to increase the text zoom to 2x, double all of the font sizes of the fonts configured for use in OpenTTD. To set the text zoom to 4x, quadruple all of the font sizes of the fonts configured for use in OpenTTD. This means the text zoom setting would be a multiplier to the font sizes specified in the OpenTTD config. If the font size in the OpenTTD config is 4px, and you set the text zoom to 4x, then the actual value used is 16px. Actual font size used in OpenTTD = font size in config * text zoom factor.

**Comment by andythenorth** [comment]
Deleted my first reply, I'll write something clearer in the morning when I'm not tired.

TL;DR reminder to self - how to fit bigger text into UI widgets without making the 'buttons and boxes' bigger (as OP asks for).  Can't see how that is ever possible TBH.

**Comment by planetmaker** [comment]
> So basically, to increase the text zoom to 2x, double all of the font sizes of the fonts configured for use in OpenTTD. To set the text zoom to 4x, quadruple all of the font sizes of the fonts configured for use in OpenTTD. This means the text zoom setting would be a multiplier to the font sizes specified in the OpenTTD config. If the font size in the OpenTTD config is 4px, and you set the text zoom to 4x, then the actual value used is 16px. Actual font size used in OpenTTD = font size in config * text zoom factor.

Seems like a valid request and by default those two could possibly be linked.

Though honestly I'm not sure it actually needs scaling in the same way.

On a side note, I always considered scaling by a factor of 2 in linear size huge steps and personally prefer smaller ones, like 1.5x (which then actually means ~twice the button area, instead of 4x which we get for the 2x scaling).

**Comment by YodasWs** [comment]
> TL;DR reminder to self - how to fit bigger text into UI widgets without making the 'buttons and boxes' bigger (as OP asks for). Can't see how that is ever possible TBH.

The buttons and boxes that contain text will of course need adjustment to fit the larger font. What I do not want changed are the icon buttons that do not have text. Nor do I want you to simply double the size of the map box (for example) just because I need the font to be twice as large.

Or maybe you guys can focus your attention and effort on the small font in the map box where the user toggles which industries to see the routes for. This is where the font is the smallest.
